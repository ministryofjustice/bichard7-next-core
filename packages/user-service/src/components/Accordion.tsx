import { ReactNode } from "react"

interface Props {
  children: ReactNode
}

const Accordion = ({ children }: Props) => (
  <div className="govuk-accordion" data-module="govuk-accordion">
    {children}
  </div>
)

export default Accordion
