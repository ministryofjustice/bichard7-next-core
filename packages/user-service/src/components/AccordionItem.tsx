import { ReactNode } from "react"

interface Props {
  children: ReactNode
  dataTest?: string
  heading: string
  id: string
}

const AccordionItem = ({ children, dataTest, heading, id }: Props) => (
  <>
    <div className="govuk-accordion__section" data-test={dataTest}>
      <div className="govuk-accordion__section-header">
        <h2 className="govuk-accordion__section-heading">
          <span className="govuk-accordion__section-button" id={id}>
            {heading}
          </span>
        </h2>
      </div>
      <div className="govuk-accordion__section-content" aria-labelledby={id}>
        {children}
      </div>
    </div>
  </>
)

export default AccordionItem
