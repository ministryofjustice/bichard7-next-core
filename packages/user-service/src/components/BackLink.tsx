import Link from "./Link"

interface Props {
  href: string
}

const BackLink = ({ href }: Props) => (
  <Link href={href} className="govuk-back-link" data-test="back-link">
    {"Back"}
  </Link>
)

export default BackLink
