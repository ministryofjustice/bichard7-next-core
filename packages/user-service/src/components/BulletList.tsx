import React from "react"

interface BulletListProps {
  heading?: string
  items: string[]
  listClassName?: string
  headingClassName?: string
}

const BulletList: React.FC<BulletListProps> = ({
  heading,
  items,
  listClassName = "govuk-list govuk-list--bullet",
  headingClassName = "govuk-body"
}) => {
  if (!items || items.length === 0) {
    return null
  }

  return (
    <>
      {heading && <p className={headingClassName}>{heading}</p>}

      <ul className={listClassName}>
        {items.map((item, index) => (
          <li key={index}>{item}</li>
        ))}
      </ul>
    </>
  )
}

export default BulletList
