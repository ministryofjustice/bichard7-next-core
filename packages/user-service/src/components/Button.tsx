import { ReactNode } from "react"
import classnames from "classnames"

type ButtonVariant = "default" | "warning" | "secondary"
interface Props {
  children: ReactNode
  variant?: ButtonVariant
  noDoubleClick?: boolean
  id?: string
  className?: string
  name?: string
  value?: string
  isDisabled?: boolean
  dataTest?: string
}

const getButtonClass = (variant?: ButtonVariant): string => {
  const variantClass = !variant || variant === "default" ? "" : ` govuk-button--${variant}`
  return `govuk-button${variantClass}`
}

const Button = ({ children, variant, noDoubleClick, id, className, name, value, isDisabled, dataTest }: Props) => {
  const classes = classnames(getButtonClass(variant), className && { [className]: !!className })

  return (
    <button
      type="submit"
      className={classes}
      name={name}
      value={value}
      data-module="govuk-button"
      data-prevent-double-click={noDoubleClick}
      data-test={dataTest}
      disabled={isDisabled}
      id={id}
    >
      {children}
    </button>
  )
}

export default Button
