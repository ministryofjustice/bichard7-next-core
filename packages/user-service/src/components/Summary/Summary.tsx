import { ReactNode } from "react"

interface Props {
  children: ReactNode
}

const Summary = ({ children }: Props) => <dl className="govuk-summary-list">{children}</dl>

export default Summary
