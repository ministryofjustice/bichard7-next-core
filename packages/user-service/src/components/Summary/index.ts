export { default as Summary } from "./Summary"
export { default as SummaryItem } from "./SummaryItem"
