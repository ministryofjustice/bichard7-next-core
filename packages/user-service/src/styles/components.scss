@import "variables";

/* B7 Inset Text */
.b7-inset-text {
  display: block;
  font-weight: 300;
  font-size: 1rem;
  padding: 0.9375rem 1.25rem;
  background: var(--other-light-grey, $b7-light-grey-color);
  box-shadow: 5px 0px 0px 0px $b7-dark-grey-color inset;

  .b7-inset-text__heading {
    display: block;
    font-weight: 700;
  }
}

/* B7 Accordion */
.b7-accordion {
  @extend .govuk-accordion__section;

  .b7-accordion__content {
    margin-top: 0.75rem;
    margin-bottom: 0.5rem;
  }

  .b7-accordion__header {
    @extend .govuk-accordion__section-header;
    margin-bottom: 0.5rem;

    .b7-accordion__heading {
      @extend .govuk-accordion__section-heading;

      .b7-accordion__button {
        @extend .govuk-accordion__section-button;

        .b7-accordion__toggle {
          @extend .govuk-accordion__section-toggle;

          .b7-accordion__toggle-focus {
            @extend .govuk-accordion__section-toggle-focus;

            .b7-accordion-nav__chevron {
              @extend .govuk-accordion-nav__chevron;
            }

            .b7-accordion-nav__chevron--down {
              @extend .govuk-accordion-nav__chevron--down;
            }

            .b7-accordion__toggle-text {
              @extend .govuk-accordion__section-toggle-text;
            }
          }
        }
      }
    }
  }
}

.js-enabled .b7-accordion .b7-accordion__header .b7-accordion__heading .b7-accordion__button,
.b7-accordion .b7-accordion__header .b7-accordion__heading .js-enabled .b7-accordion__button {
  padding: 0;
  border: none;
}

.js-enabled .b7-accordion .b7-accordion__header .b7-accordion__heading .b7-accordion__button .b7-accordion__toggle,
.b7-accordion .b7-accordion__header .b7-accordion__heading .b7-accordion__button .js-enabled .b7-accordion__toggle {
  margin-bottom: 0;
}

/* MoJ badge */
.moj-badge {
  background-color: white;
}

.preview-button {
  margin-Bottom: 0px;
  white-space: nowrap;
}