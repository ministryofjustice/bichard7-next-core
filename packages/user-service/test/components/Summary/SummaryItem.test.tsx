/* eslint-disable @typescript-eslint/no-non-null-assertion */
import { render } from "@testing-library/react"
import { SummaryItem } from "components/Summary"

it("should render component", () => {
  const { container } = render(<SummaryItem label="Label name" value="Label value" />)

  expect(container).toMatchSnapshot()
})
