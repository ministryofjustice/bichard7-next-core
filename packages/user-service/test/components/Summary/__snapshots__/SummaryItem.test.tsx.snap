// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`should render component 1`] = `
<div>
  <div
    class="govuk-summary-list__row"
  >
    <dt
      class="govuk-summary-list__key"
      data-test="summary-item_undefined_label"
    >
      Label name
    </dt>
    <dd
      class="govuk-summary-list__value"
      data-test="summary-item_undefined_value"
    >
      Label value
    </dd>
  </div>
</div>
`;
